phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -774,7 +782,12 @@ impl TestVm {
 
     async fn send_serial_bytes_async(&self, bytes: Vec<u8>) -> Result<()> {
         match &self.state {
-            VmState::Ensured { serial } => serial.send_bytes(bytes),
+            VmState::Ensured { serial } => {
+                let (done_tx, done_rx) = oneshot::channel();
+                serial.send_bytes(bytes, done_tx)?;
+                done_rx.await?;
```

Comment:
nit, take it or leave it: i _might_ move the oneshot into `send_bytes`, and make it an `async fn` that creates a oneshot, sends the `tx` end to the serial worker, and then awaits the `rx` end? that way, here, we could just write `serial.send_bytes(bytes).await`...

the big reason _not_ to do that is if we want to be able to call `send_bytes()` in a non-async context,[^1] which ... i don't _think_ we do anywhere? but maybe we want to be able to do it later? 

[^1]: and either pass the `rx` end of the oneshot into something that eventually awaits it in an async context, or just throw it away because we don't care.

Reply:
actually, thinking about this some more, if `send_bytes` was a non-async function that just returned the `rx`, we could write `serial.send_bytes(bytes)?.await?;` here, but still retain the optionality to call it without immediately awaiting and pass the oneshot `rx` to someone else/chuck it in the garbage immediately/etc.

Reply:
Done in b5b8d81 (along with some general cleanup of this whole family of TestVm functions).